aten/src/ATen/native/cuda/CuFFTPlanCache.h

```diff
@@ -254,6 +298,19 @@ class CuFFTConfig {
       }
 
       // set odist, onembed, base_ostride
+#ifdef __HIP_PLATFORM_HCC__
+      int odist = at::prod_intlist(output_sizes.slice(1, signal_ndim));
+      std::vector<int> onembed(output_sizes.data() + 1, output_sizes.data() + signal_ndim + 1);
+      int base_ostride = 1;
+
+      int sizes = *signal_sizes.data();
+      int istride = base_istride;
+      int iidist = idist;
+      CUFFT_CHECK(hipfftMakePlanMany(plan(), signal_ndim, &sizes,
```

Comment:
Uhhhh. Unless `signal_ndim` is always 1, I am **highly skeptical** that this code works. Aren't you going to buffer overflow the `&sizes` address.

Reply:
good catch, thanks!